core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -156,8 +158,12 @@ public void put(ClusterService clusterService, PutPipelineRequest request, Actio
             // validates the pipeline and processor configuration before submitting a cluster update task:
             Map<String, Object> pipelineConfig = XContentHelper.convertToMap(request.getSource(), false).v2();
             constructPipeline(request.getId(), pipelineConfig);
+        } catch (ConfigurationPropertyException e) {
```

Comment:
since this logic is only really necessary in `put` to validate pipeline _can_ be constructed, I feel like we can rename `constructPipeline` to something more like a pipeline verification utility, including all this exception logic